### PR TITLE
Update the README

### PR DESCRIPTION
**Pull request by ptpaterson**
Ticket(s): [FE-3173](https://faunadb.atlassian.net/browse/FE-3173)

## Problem
Need to update the README for publishing on NPM.

## Solution
Content has been reordered and reworked. Left some details out that are more in flux (like errors) and details that will be included in the documentation (like the wire protocol).

----
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

[FE-3173]: https://faunadb.atlassian.net/browse/FE-3173?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ptpaterson** [comment]
Left as a draft until I 

- add back sections for client and query configuration.
- copy changes to the demo

File: README.md [hunk 1]
```diff
@@ -1,322 +1,329 @@
...
+- Firefox
+- Safari
+- Edge
 
-Default settings are available for the client. So you can creat a client as simply as:
+# Installation
 
-```typescript
-import { Client } from "fauna";
+## Node.js
```
Comment (cleve-fauna) [inline README.md:54]: Node.js feels misleading.

You don't have to be executing on node to install with npm or yarn.

I'd call this section ## Installing

```suggestion
## Installing
The fauna-js driver is available on npm.

You can install with your package manager of choice:

* `npm install fauna`
* `yarn add fauna`
```

File: README.md [hunk 2]
```diff
@@ -1,322 +1,329 @@
...
-  } catch (e) {
-    console.log(e);
-  }
-})();
+```shell
+yarn add fauna
 ```
 
-## Export your secret as an environment variable
+## Browsers
```
Comment (cleve-fauna) [inline README.md:66]: This feels like an edge case.

Calling it "browsers" feels a little odd as if i'm writing code to run in a browser i'm probably going to use an actual package manager and not whack the driver straight into my script.

```suggestion
## Available via CDN
```

File: README.md [hunk 3]
```diff
@@ -1,322 +1,329 @@
...
 
-Rather than hard coding your secret into your code, the driver can load your secret automatically from a FAUNA_SECRET environment variable. Open a terminal and execute the following command:
+The driver is built as a JS module for browsers and is available via CDN:
```
Comment (cleve-fauna) [inline README.md:68]:
```suggestion
The driver is additionally made available to browsers via CDN:
```

File: README.md [hunk 4]
```diff
@@ -1,322 +1,329 @@
...
-You can install with yarn with `yarn add fauna`
+<details>
+<summary>Table of Contents</summary>
 
-## Pre-requisites
+- [A JavaScript driver for Fauna.](#a-javascript-driver-for-fauna)
+- [Compatibility](#compatibility)
+- [Installation](#installation)
+  - [Node.js](#nodejs)
+  - [Browsers](#browsers)
```
Comment (cleve-fauna) [inline README.md:20]: IMO having quick starts at the tops of READMEs makes getting ramped up on a new dependency much easier.

I'd suggest leading with a quick start that includes installation and basic usage. And moving compatibility below that.

File: README.md [hunk 5]
```diff
@@ -1,322 +1,329 @@
...
-```typescript
-const greetee = "World!";
-const result = await client.query<string>(fql`"Hello " + ${greetee}`);
-console.log(result);
-// prints { data: "Hello World!", txn_ts: 1667174400000000 } // txn_ts will vary!
+try {
+  // create a Collection
+  const collection_query = fql`Collection.create({ name: "Dogs" })`;
+  const collection_result = await client.query(collection_query);
+
```
Comment (cleve-fauna) [inline README.md:120]: i'd show some basic interpolation here. Perhaps put "Dogs" in a variable

File: README.md [hunk 6]
```diff
@@ -1,322 +1,329 @@
...
-// of a larger query.
-function collectionExists(collectionName: string) {
-  return fql`Collection.byName(${collectionName}) != null`;
-}
+Here's an example in pure FQL X:
 
-let collectionName = "ComposedQuery";
-// below, we compose a FQL X query that injects the FQL X
-// generated by collectionExists
+```javascript
```
Comment (cleve-fauna) [inline README.md:145]: this example feels pretty heavyweight IMO.

Its not easy to read at a glance, and you have to grok that its actually building an anonymous function in _FQL_ itself to understand how it works.

I'd keep it simple here to show the basic power.

Complex examples like this are going to make it difficult to understand what's going on.

The previous example met that simplicity criteria IMO.

Comment (ptpaterson) [reply README.md:148]: This came out of wanting to use a diverse example for the composition query. There we have 3 different kinds of composition:

- pure fql snippet
- reusable projection (similar to GraphQL fragments)
- inserting data

The pure fql query is then the exact same as the composition query.

I started off with a different pure fql example and had this example under query composition to explain what the composed one was doing. How does that sound to you?

**Comment by cleve-fauna** [comment]
Let's make sure the README also accounts for #107 where we changed the endpoints provided as defaults.

File: README.md (hunk 6 above)
Comment (ptpaterson) [reply README.md]: Quickstart and configuration examples are pretty good demonstrations at a glance. I wrapped up these sections into a new "Advanced Example" section at the bottom.

**Pull request merged**